lock/src/main/java/com/auth0/android/lock/utils/EmailParser.java

```diff
@@ -0,0 +1,150 @@
+/*
+ * DomainParser.java
+ *
+ * Copyright (c) 2016 Auth0 (http://auth0.com)
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package com.auth0.android.lock.utils;
+
+import android.support.annotation.Nullable;
+
+import com.auth0.Auth0Exception;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Helper class to find out which email domains can be valid for the current Auth0 configuration.
+ */
+public class EmailParser {
+
+    private static final String DOMAIN_KEY = "domain";
+    private static final String DOMAIN_ALIASES_KEY = "domain_aliases";
+
+    private List<Strategy> strategies;
+
+    public EmailParser(List<Strategy> strategies) {
+        if (strategies == null || strategies.isEmpty()) {
+            throw new Auth0Exception("You must provide a valid list of Strategies.");
```

Comment:
Let's avoid this kind of behaviour whenever possible since it will crash the app unless we are catching this kind of exception (which I doubt). If there is no enterprise strategies we should always return no connection.